source/CellMeasurer/CellMeasurerCache.js

```diff
@@ -225,18 +225,29 @@ export default class CellMeasurerCache {
     // :columnWidth and :rowHeight are derived based on all cells in a column/row.
     // Pre-cache these derived values for faster lookup later.
     // Reads are expected to occur more frequently than writes in this case.
+    const columnKey = this._keyMapper(0, columnIndex)
+    const rowKey = this._keyMapper(rowIndex, 0)
     let columnWidth = 0
-    for (let i = 0; i < this._rowCount; i++) {
-      columnWidth = Math.max(columnWidth, this.getWidth(i, columnIndex))
+    if (this._hasFixedWidth) {
+      // if has fixed width columns, no need to calculate width for each row
```

Comment:
If the hasFixedWidth is true then cell measurer cache can only use defaultWidth. Or you can substitute this block with getWidth function. Look at the implementation of getWidth it is basicly doing the same thing as you in this block of code.
```javascript
columnWidth = this.defaultWidth;
// or
columnWidth = this.getWidth(rowIndex, columnIndex);
```